packages/core/test/app/ovm/deciders/for-all-such-that-decider.spec.ts

```diff
@@ -0,0 +1,499 @@
+import '../../../setup'
+
+import {
+  ForAllSuchThatDecider,
+  ForAllSuchThatInput,
+  CannotDecideError,
+} from '../../../../src/app/ovm/deciders'
+import { CannotDecideDecider, FalseDecider, TrueDecider } from './utils'
+import {
+  Decision,
+  Property,
+  PropertyFactory,
+  Quantifier,
+  QuantifierResult,
+  WitnessFactory,
+} from '../../../../src/types/ovm'
+import * as assert from 'assert'
+
+/*******************
+ * Mocks & Helpers *
+ *******************/
+
+class DummyQuantifier implements Quantifier {
+  public async getAllQuantified(parameters: any): Promise<QuantifierResult> {
+    return undefined
+  }
+}
+
+const getQuantifierThatReturns = (
+  results: any[],
+  allResultsQuantified: boolean
+): Quantifier => {
+  const quantifier: Quantifier = new DummyQuantifier()
+  quantifier.getAllQuantified = async (params): Promise<QuantifierResult> => {
+    return {
+      results,
+      allResultsQuantified,
+    }
+  }
+  return quantifier
+}
+
+const getPropertyFactoryThatReturns = (
+  properties: Property[]
+): PropertyFactory => {
+  return (input: any): Property => {
+    return properties.shift()
+  }
+}
+
+const undefinedWitnessFactory = (input: any) => undefined
+
+/*********
+ * TESTS *
+ *********/
+
+describe('ForAllSuchThatDecider', () => {
+  let decider: ForAllSuchThatDecider
+
+  const trueDecider: TrueDecider = new TrueDecider()
+  const falseDecider: FalseDecider = new FalseDecider()
+  const cannotDecideDecider: CannotDecideDecider = new CannotDecideDecider()
+  beforeEach(() => {
+    decider = new ForAllSuchThatDecider()
+  })
+
+  const testReturnTrueWithNoDeciders = async (isDecide: boolean = true) => {
```

Comment:
Hmm so I feel this testing pattern works great for our current conception of deciders with `decide` and `checkDecision`. However, I feel that generally it would be nice if our model was simplified and I think it can be.

Tomorrow on standup maybe it would be useful for us to jam a bit on how this could (maybe?) be simplified. There seems to be a little bit of cruft that we can remove that would remove one of these logical branches. I think at this point it’s shown that having two logical branches is adding complexity that is already starting to add up a bit. Plus I think it could be useful to discuss the concept of caching synchronously.